### PR TITLE
Show editions table w/o limit if < 10 editions

### PR DESCRIPTION
**Pull request by mekarpeles**
Use edition_count to ignore limit and show all books if total editions_count < 10.

### Technical

Moved editions_count up (which hits solr). Can this ever fail? When it does, edition_count is set to 1. Can this cause problems?

### CODE REVIEW

File: openlibrary/templates/type/edition/view.html [hunk 1]
```diff
@@ -28,11 +28,12 @@
...
 $# Book availability of editions injected by bulk get_availability API
+$ edition_count = work.edition_count if work and work.edition_count else 1
 $ ebooks_only = query_param('mode') != 'all'
 
-$# For performance reasons, only load 10
-$# Tradeoff: decreases/limits our ability to select best edition
-$ editions_limit = 10 if not query_param('mode') else None
+$# For performance reasons, limit to 10 ebooks
+$# Tradeoff: limits our ability to select best edition
+$ editions_limit = 10 if not query_param('mode') and edition_count <= 10 else None
```
Comment (mekarpeles) [inline openlibrary/templates/type/edition/view.html:36]:
```suggestion
$ editions_limit = 10 if not query_param('mode') and edition_count > 10 else None
```

File: openlibrary/templates/type/edition/view.html [hunk 2]
```diff
@@ -28,11 +28,12 @@
...
+$ edition_count = work.edition_count if work and work.edition_count else 1
 $ ebooks_only = query_param('mode') != 'all'
 
-$# For performance reasons, only load 10
-$# Tradeoff: decreases/limits our ability to select best edition
-$ editions_limit = 10 if not query_param('mode') else None
+$# For performance reasons, limit to 10 ebooks
+$# Tradeoff: limits our ability to select best edition
+$ editions_limit = 10 if not query_param('mode') and edition_count > 10 else None
 $ editions = work.get_sorted_editions(ebooks_only=ebooks_only, limit=editions_limit, keys=edition and [edition.key])
```
Comment (mekarpeles) [inline openlibrary/templates/type/edition/view.html:37]:
```suggestion
$if editions_limit:
  $ editions = work.get_sorted_editions(ebooks_only=ebooks_only, limit=editions_limit, keys=edition and [edition.key])
```

File: openlibrary/templates/type/edition/view.html [hunk 3]
```diff
@@ -28,12 +28,14 @@
...
 $ ebooks_only = query_param('mode') != 'all'
 
-$# For performance reasons, only load 10
-$# Tradeoff: decreases/limits our ability to select best edition
-$ editions_limit = 10 if not query_param('mode') else None
-$ editions = work.get_sorted_editions(ebooks_only=ebooks_only, limit=editions_limit, keys=edition and [edition.key])
+$# For performance reasons, limit to 10 ebooks
+$# Tradeoff: limits our ability to select best edition
+$ editions_limit = 10 if not query_param('mode') and edition_count > 10 else None
+$if editions_limit:
```
Comment (mekarpeles) [inline openlibrary/templates/type/edition/view.html:37]:
```suggestion
$if editions_limit or query_param('mode') == 'ebooks':
```

**Review by cdrini — COMMENTED** [review]
- ✅ show non-borrowable as well https://testing.openlibrary.org/works/OL2070372W/Life_with_Picasso?edition=ia%3Alifewithpicasso100gilo
- ✅ shows all 10 https://testing.openlibrary.org/works/OL3901899W/Cases_in_constitutional_law?edition=ia%3Aunset0000unse_z6z9
- ✅ shows only IA if >10 eds https://testing.openlibrary.org/works/OL4943454W/The_history_of_the_discovery_and_settlement_of_America?edition=ia%3Ahistdiscoverysett00roberich
- ❌ Shows only 10 IA editions https://testing.openlibrary.org/works/OL9170454W/Hamlet?edition=ia%3Atragedyofhamletp00shak
- ✅ clicking view all works https://testing.openlibrary.org/works/OL9170454W/Hamlet?mode=all

**Review by cdrini — APPROVED** [review]
Ah, looks like you're right @mekarpeles , it's only TBP editions. LGTM!

**Pull request merged**